pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -419,17 +419,23 @@ func (ca *changeAggregator) setupSpansAndFrontier(
 		return nil, err
 	}
 
-	// Checkpointed spans are spans that were above the highwater mark, and we
-	// must preserve that information in the frontier for future checkpointing.
 	// If we don't have a highwater yet (during initial scan) they must at least
 	// be from StatementTime, and given an initial highwater they must all by
 	// definition have been at or after initialHighWater.Next()
-	var checkpointedSpanTs hlc.Timestamp
-	if initialHighWater.IsEmpty() {
-		checkpointedSpanTs = ca.spec.Feed.StatementTime
-	} else {
-		checkpointedSpanTs = initialHighWater.Next()
+	checkpointedSpanTs := ca.spec.Checkpoint.Timestamp
```

Comment:
I just realized that just having frontier forwarded to this timestamp doesn't actually do much for us.
If we checkpointed due to slow spans (i.e. no table events, backfills, etc), then ... we would still resume the rangefeed for all the spans from the checkpointed timestamps...